File: Level 1/Week01/Day1/hhs/2.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        int[] alphabets = new int[26];
        Arrays.fill(alphabets, -1);

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);

            answer[i] = (alphabets[ch - 'a'] == -1) ? -1 : i - alphabets[ch - 'a'];
            alphabets[ch - 'a'] = i;
        }
        return answer;
    }
}

File: Level 2/Week04/Day25/hhs/2.kt
package pro_142085

import java.util.Collections
import java.util.PriorityQueue

class Solution {
    fun solution(n: Int, k: Int, enemy: IntArray): Int {
        var answer = 0
        val pq: PriorityQueue<Int> = PriorityQueue(Collections.reverseOrder())

        var mySoldier = n
        var card = k
        for (e in enemy) {
            pq.add(e)
            mySoldier -= e

            if (mySoldier < 0) {
                if (card > 0) {
                    mySoldier += pq.poll()
                    card--
                } else {
                    break
                }
            }
            answer++
        }

        return answer
    }
}

fun main() {
    val s = Solution()
    println(s.solution(7, 3, intArrayOf(4, 2, 4, 5, 3, 3, 1)))
}